ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -36,56 +49,255 @@ func FaultNetworkFaultPath(fault string) string {
+		requestType := fmt.Sprintf(checkStatusFaultRequestType, types.BlackHolePortFaultType)
+
+		err := decodeRequest(w, &request, requestType, r)
+		if err != nil {
+			return
+		}
+		// Validate the fault request
+		err = validateRequest(w, request, requestType)
+		if err != nil {
+			return
+		}
+		logger.Debug("Successfully parsed fault request payload", logger.Fields{
+			"payload": request.ToString(),
+		})
+
+		// Obtain the task metadata via the endpoint container ID
+		// TODO: Will be using the returned task metadata in a future PR
+		_, err = getTaskMetadataForFault(w, h.AgentState, requestType, r)
+		if err != nil {
+			return
+		}
+
+		// Check status of current fault injection
+
+		logger.Debug("Successfully checked status for fault", logger.Fields{
+			"Request Type": requestType,
+		})
+		// TODO: Return the correct status state
+		utils.WriteJSONResponse(
+			w,
+			http.StatusOK,
+			types.NetworkFaultInjectionResponse{
+				Status: "running",
+			},
+			requestType,
+		)
 	}
 }
 
 // TODO
-func (h *FaultHandler) StartLatency() func(http.ResponseWriter, *http.Request) {
+func (h *FaultHandler) StartNetworkLatency() func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
+
 	}
 }
 
 // TODO
-func (h *FaultHandler) StopLatency() func(http.ResponseWriter, *http.Request) {
+func (h *FaultHandler) StopNetworkLatency() func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
 // TODO
-func (h *FaultHandler) CheckLatencyStatus() func(http.ResponseWriter, *http.Request) {
+func (h *FaultHandler) CheckNetworkLatency() func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
 // TODO
-func (h *FaultHandler) StartPacketLoss() func(http.ResponseWriter, *http.Request) {
+func (h *FaultHandler) StartNetworkPacketLoss() func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
 // TODO
-func (h *FaultHandler) StopPacketLoss() func(http.ResponseWriter, *http.Request) {
+func (h *FaultHandler) StopNetworkPacketLoss() func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
 // TODO
-func (h *FaultHandler) CheckPacketLossStatus() func(http.ResponseWriter, *http.Request) {
+func (h *FaultHandler) CheckNetworkPacketLoss() func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 	}
 }
+
+func decodeRequest(w http.ResponseWriter, request types.NetworkFaultRequestInterface, requestType string, r *http.Request) error {
+	jsonDecoder := json.NewDecoder(r.Body)
+	jsonDecoder.DisallowUnknownFields()
+	if err := jsonDecoder.Decode(request); err != nil {
+		logger.Error("Error: failed to decode request", logger.Fields{
+			field.Error:   err,
+			"requestType": requestType,
+			"body":        request,
+		})
+
+		utils.WriteJSONResponse(
```

Comment:
I think we should log the response.

Reply:
Yes, I made these changes before our offline talk about this. This comment and all of the other ones have been address.